lib/install/management/delete.go

```diff
@@ -15,41 +15,75 @@
 package management
 
 import (
+	"fmt"
 	"strings"
 
 	log "github.com/Sirupsen/logrus"
 
+	"github.com/vmware/govmomi/vim25/types"
 	"github.com/vmware/vic/lib/metadata"
 	"github.com/vmware/vic/pkg/errors"
 	"github.com/vmware/vic/pkg/vsphere/compute"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
 	"github.com/vmware/vic/pkg/vsphere/vm"
+
+	"golang.org/x/net/context"
 )
 
 func (d *Dispatcher) DeleteVCH(conf *metadata.VirtualContainerHostConfigSpec) error {
-	log.Infof("Removing VMs")
+	var errs []string
+
+	var err error
+	var vmm *vm.VirtualMachine
 
-	if err := d.DeleteVCHInstances(conf); err != nil {
+	if vmm, err = d.findApplianceByID(conf); err != nil {
 		return err
 	}
-	if err := d.destroyResourcePoolIfEmpty(conf); err != nil {
+	if vmm == nil {
+		return nil
+	}
+
+	if err = d.DeleteVCHInstances(vmm, conf); err != nil {
+		// if container delete failed, do not remove anything else
+		log.Infof("Specify --force to force delete")
 		return err
 	}
-	return d.removeNetwork(conf)
+
+	if err = d.DeleteDataStores(vmm, conf); err != nil {
+		errs = append(errs, err.Error())
+	}
+
+	if err = d.deleteNetworkDevices(vmm, conf); err != nil {
```

Comment:
Is it actually necessary to delete the network devices before deleting the network? I may be recalling incorrectly but I thought it was possible to do so and associated NICs would simply become disconnected.